### PR TITLE
[14.0][ADD] purchase_order_vendor_products_only

### PR DESCRIPTION
**Pull request by ivs-cetmix**
Limit selection of the products in a PO to the ones available from the selected Venor

### CODE REVIEW

**Review by elvise — APPROVED** [review]
LGTM

**Comment by elvise** [comment]
@OCA/community-maintainers take a look

**Review by francesco-ooops — APPROVED** [review]
Functional review ok

**Review by HviorForgeFlow — CHANGES_REQUESTED** [review]
Could you please do these changes:
- I would name it like `purchase_order_restrict_vendor_product` (Avoiding plurals in module name)
 - Then change commit description to [14.0][ADD] purchase_order_restrict_vendor_product
 - I think the PR needs a rebase or at least remove changes in setup/_metapackage

**Review by simahawk — CHANGES_REQUESTED** [review]
@isserver1 thanks for your contrib.
You should fix this by:
* rebasing on 14.0
* do not delete any exiting file
* cherry-pick your own commits
* rewrite your commits as `Add purchase_order_vendor_product` and `Add purchase_order_vendor_product_only`

If you prefer you can open another PR starting again from 14.0

**Comment by simahawk** [comment]
```
2022-04-15 13:02:09,764 8874 INFO openerp_test odoo.modules.loading: Module purchase_picking_state: loading demo
2022-04-15 13:02:09,904 8874 INFO openerp_test unittest.suite: ======================================================================
2022-04-15 13:02:09,905 8874 ERROR openerp_test unittest.suite: ERROR: setUpClass (odoo.addons.purchase_picking_state.tests.test_purchase_picking_state.TestPurchasePickingState)
Traceback (most recent call last):
File "/home/travis/odoo-14.0/odoo/tests/common.py", line 173, in _handleClassSetUp
setUpClass()
File "/home/travis/odoo-14.0/addons/account/tests/common.py", line 27, in setUpClass
assert 'post_install' in cls.test_tags, 'This test requires a CoA to be installed, it should be tagged "post_install"'
AssertionError: This test requires a CoA to be installed, it should be tagged "post_install"
```

**Comment by ivs-cetmix** [comment]
> ```
> 2022-04-15 13:02:09,764 8874 INFO openerp_test odoo.modules.loading: Module purchase_picking_state: loading demo
> 2022-04-15 13:02:09,904 8874 INFO openerp_test unittest.suite: ======================================================================
> 2022-04-15 13:02:09,905 8874 ERROR openerp_test unittest.suite: ERROR: setUpClass (odoo.addons.purchase_picking_state.tests.test_purchase_picking_state.TestPurchasePickingState)
> Traceback (most recent call last):
> File "/home/travis/odoo-14.0/odoo/tests/common.py", line 173, in _handleClassSetUp
> setUpClass()
> File "/home/travis/odoo-14.0/addons/account/tests/common.py", line 27, in setUpClass
> assert 'post_install' in cls.test_tags, 'This test requires a CoA to be installed, it should be tagged "post_install"'
> AssertionError: This test requires a CoA to be installed, it should be tagged "post_install"
> ```

Yes, I saw this. But cannot understand how is this connected with the module I add. Would appreciate your assistance

File: purchase_advance_payment/__manifest__.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 {
     "name": "Purchase Advance Payment",
-    "version": "14.0.1.0.0",
+    "version": "14.0.1.0.1",
```
Comment (simahawk) [inline purchase_advance_payment/__manifest__.py]: are you sure you've rebased properly? These changes should not be here apparently.

**Review by simahawk — CHANGES_REQUESTED** [review]

**Comment by simahawk** [comment]
> > ```
> > 2022-04-15 13:02:09,764 8874 INFO openerp_test odoo.modules.loading: Module purchase_picking_state: loading demo
> > 2022-04-15 13:02:09,904 8874 INFO openerp_test unittest.suite: ======================================================================
> > 2022-04-15 13:02:09,905 8874 ERROR openerp_test unittest.suite: ERROR: setUpClass (odoo.addons.purchase_picking_state.tests.test_purchase_picking_state.TestPurchasePickingState)
> > Traceback (most recent call last):
> > File "/home/travis/odoo-14.0/odoo/tests/common.py", line 173, in _handleClassSetUp
> > setUpClass()
> > File "/home/travis/odoo-14.0/addons/account/tests/common.py", line 27, in setUpClass
> > assert 'post_install' in cls.test_tags, 'This test requires a CoA to be installed, it should be tagged "post_install"'
> > AssertionError: This test requires a CoA to be installed, it should be tagged "post_install"
> > ```
> 
> Yes, I saw this. But cannot understand how is this connected with the module I add. Would appreciate your assistance

Seems to be a conflict between modules. As you see here https://github.com/OCA/purchase-workflow/blob/14.0/.travis.yml#L19 certain modules run their tests w/out others.
Your modules - in theory - are not adding any functional changes that might hinder tests but you pulled other changes that might cause it. Just wild guessing...

**Pull request closed without merging**